plugin/proto/metadata.go

```diff
@@ -42,6 +45,12 @@ func (resp *GetMetadataResponse) Validate() error {
 	if len(resp.SupportedContractVersions) == 0 {
 		return errors.New("empty supported contract versions")
 	}
+	if !resp.SupportsContract(ContractVersion) {
```

Comment:
Should we unexport `SupportsContract()` as the contract version set and check are encapsulated in the `plugin`, `plugin/proto` packages?

Reply:
moved SupportsContract to plugin/plugin.go supportsContract()